common/VotingUtils.js

```diff
@@ -51,14 +55,18 @@ const constructCommitment = async (request, roundId, web3, price, account) => {
 
 /**
  * Decrypt an encrypted vote commit for the voter and return vote details
- * @param {* Object} request {identifier, time}
- * @param {* String} roundId
- * @param {* Object} web3
- * @param {* String} account
- * @param {* Object} votingContract deployed Voting.sol instance
+ * @param {Object} request {identifier, time}
+ * @param {String} roundId
+ * @param {Object} web3
+ * @param {String} account account to sign message with
+ * @param {Object} votingContract deployed Voting.sol instance
+ * @param {Object} [designatedVotingAddress] proxy voting contract,
  */
-const constructReveal = async (request, roundId, web3, account, votingContract) => {
-  const encryptedCommit = (await getLatestEvent("EncryptedVote", request, roundId, account, votingContract))
+const constructReveal = async (request, roundId, web3, account, votingContract, designatedVotingAddress) => {
+  // Voting account is the "voter" whose past votes we query to find encrypted commits.
+  const votingAccount = designatedVotingAddress ? designatedVotingAddress : account;
```

Comment:
Note: I don't think we necessarily need to do this now.

We perform this same if statement in quite a few places to determine two distinct things:
1. What contract we should send the vote calls to.
2. What address we should use as the voter.

It could make sense to have a library that constructs an object that has three properties:
1. `votingContract`: where to send the votes.
2. `votingAccount`: address that the votes are attributed to.
3. `signingAddress`: address used to sign the encrypted messages.

Then, as long as the correct one is accessed, we don't have to worry about computing it incorrectly in some places or accidentally assuming that the user does or doesn't use a designated voting contract.

Just a thought.

Reply:
I like this, it avoids refactoring the `VotingUtils` libraries

Reply:
Done